### ISSUE TITLE
Minimal monitoring support

### ISSUE THREAD

**Issue by quentinblampey** [snakemake/snakemake#2260]
Hello,

I'm running a snakemake pipeline on a slurm HPC for dozens of similar datasets. The same pipeline is applied on one single dataset, so I sometimes have up to ten pipelines running in parallel.

### Aim
My problem is very simple: I want to check which pipelines are currently running, which slurm jobs were submitted, and what is the status of these running jobs (e.g., pending/running/completed).

### What I'm currently doing
For now, I'm creating an additionnal log files for each pipeline to tell which are running or not. I don't really have a better/clean way to find which snakemake pipeline are running.

Then, I made a script that:
- parse the log files of the currently running pipeline
- grab the IDs of the submitted jobs
- run "seff <jobid>" to have some basic info about these running jobs
- summarise all this info in a table.

This is very dirty, and also very time-consuming because parsing long log files can take dozens of seconds for each pipeline run.

### Summary
I think I could improve my little monitoring system, but maybe a snakemake feature for that could be great? It's a very simple task, but that is very informative: without that, I'm completely blind to what is currently running, because sometimes I have 200 running jobs in parallel and it's not easy to check which of these jobs are running on which dataset.

**Comment by cademirch** [comment]
Yeah updated monitoring would be nice. I know it did/does [exist].(https://snakemake.readthedocs.io/en/stable/executing/monitoring.html) But I don't think its dependency, [panoptes](https://github.com/panoptes-organization/panoptes) is being maintained anymore? Have you tried this to see if it works? I would be interested in contributing to this if it needs work.

**Comment by quentinblampey** [comment]
Hello @cademirch, thanks for your quick answer.
I saw this panoptes option, but, as you mentioned, it doesn't seem to be maintained anymore. Actually, I didn't tried yet, because the snakemake doc was talking about "future releases", and the doc currently doesn't provide a lot of details about panoptes usage.

Also, my pipeline runs on a HPC, so I guess that I will need to make a SSH tunnel to access the Flask app?

I believe that a simple CLI with some options and a nice console display would have been enough, but I will try panoptes and see if it works and suits my needs :)

**Comment by cademirch** [comment]
Sounds good. Looks like there was recent activity in panoptes issues, so I think the developers are still around. 

Though a textual report/overview of the running workflow could be cool and useful.

**Comment by quentinblampey** [comment]
Hello @cademirch,

I finally took some time to test Panoptes, sorry for the delay.
It doesn't provide much information right now: it only shows the status of each pipeline (e.g., 'running' or 'done') and the duration of each job. It's quite helpful, but I was expecting a few more features, for instance, showing the RAM usage, and the possibility to add more metadata in the main workflow table (so that we can better search inside the list of pipeline runs).

For now, I will continue with my current handmade monitoring CLI :)

**Comment by cademirch** [comment]
@quentinblampey Sounds good. If you are willing to share your CLI, I would be interested in trying it out!

**Comment by quentinblampey** [comment]
The CLI that I made is very specific to one of my (private) project, so it will not work for other pipelines, unfortunately...But I'm willing to share as much as I can, so I made this [snippet repository](https://github.com/quentinblampey/smk_cli_snippet) with some minimal functionalities. Note that I used [Bashly](https://bashly.dannyb.co/), it is a very nice feature-rich-CLI generator!

For now, this CLI is not very mature and not well documented, so I hope you will be able to find still some useful code that you could try. Maybe in the future, I will have something cleaner, more documented, and general to all pipelines on slurm: in this case I will of course share the complete version :)
